contrib/camel/pkg/controller/reconcile.go

```diff
@@ -0,0 +1,323 @@
+	}
+
+	camelSource, err := r.buildSourceCode(ctx, source)
+	if err != nil {
+		r.recorder.Eventf(source, corev1.EventTypeWarning, "SourceCodeFailed", "Failed to build camel source: %v", err)
+		return object, err
+	}
+
+	args := &resources.CamelArguments{
+		Name:               source.Name,
+		Namespace:          source.Namespace,
+		Source:             camelSource,
+		ServiceAccountName: source.Spec.ServiceAccountName,
+		Context:            integrationContextName,
+		Sink:               sinkURI,
+	}
+
+	integration, err := r.getIntegration(ctx, source)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			integration, err = r.createIntegration(ctx, source, args)
+			if err != nil {
+				r.recorder.Eventf(source, corev1.EventTypeWarning, "IntegrationBlocked", "waiting for %v", err)
+				return object, err
+			}
+			r.recorder.Eventf(source, corev1.EventTypeNormal, "Deployed", "Created integration %q", integration.Name)
+			source.Status.MarkDeploying("Deploying", "Created integration %s", integration.Name)
+			// Since the Deployment has just been created, there's nothing more
+			// to do until it gets a status. This CamelSource will be reconciled
+			// again when the Integration is updated.
+			return object, nil
+		}
+		return object, err
+	}
+
+	// Update Integration spec if it's changed
+	expected, err := resources.MakeIntegration(args)
+	if err != nil {
+		return object, err
+	}
+	// Since the Integration spec has fields defaulted by the webhook, it won't
+	// be equal to expected. Use DeepDerivative to compare only the fields that
+	// are set in expected.
+	if !equality.Semantic.DeepDerivative(expected.Spec, integration.Spec) {
+		logger.Infof("Integration %q in namespace %q has changed and needs to be updated", integration.Name, integration.Namespace)
+		integration.Spec = expected.Spec
+		err := r.client.Update(ctx, integration)
+		// if no error, update the status.
+		if err == nil {
+			r.recorder.Eventf(source, corev1.EventTypeNormal, "Deployed", "Updated integration %q", integration.Name)
+			source.Status.MarkDeploying("IntegrationUpdated", "Updated integration %s", integration.Name)
+		} else {
+			source.Status.MarkDeploying("IntegrationNeedsUpdate", "Attempting to update integration %s", integration.Name)
+			r.recorder.Eventf(source, corev1.EventTypeWarning, "IntegrationNeedsUpdate", "Failed to update integration %q", integration.Name)
+		}
+		// Return after this update or error and reconcile again
+		return object, err
+	}
+
+	// Create the platform if not present
+	if err := r.syncPlatform(ctx, source.Namespace); err != nil {
+		logger.Error("Error occurred when trying to sync the camel platform")
+		return object, err
+	}
+
+	// Update source status
+	if integration.Status.Phase == camelv1alpha1.IntegrationPhaseRunning {
+		source.Status.MarkDeployed()
+	}
+
+	return source, nil
+}
+
+func (r *reconciler) buildSourceCode(ctx context.Context, source *v1alpha1.CamelSource) (resources.CamelArgumentsSource, error) {
```

Comment:
I don't see any reconciler arguments being used here, it looks like this should move to the resources dir.